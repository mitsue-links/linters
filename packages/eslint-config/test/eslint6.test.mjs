import path from 'path';
import {lintFile} from './helper.mjs';

describe('ESLint 6.2.0', () => {
    test('6.2.0_ok.mjs', async () => {
        const filePath = '6.2.0_ok.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toStrictEqual([{
            filePath,
            messages: [],
        }]);
    });

    test('6.2.0_error.mjs', async () => {
        const filePath = '6.2.0_error.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toMatchSnapshot();
    });
});

describe('ESLint 6.4.0', () => {
    test('6.4.0_ok.mjs', async () => {
        const filePath = '6.4.0_ok.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toStrictEqual([{
            filePath,
            messages: [],
        }]);
    });

    test('6.4.0_error.mjs', async () => {
        const filePath = '6.4.0_error.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toMatchSnapshot();
    });
});

describe('ESLint 6.5.0', () => {
    test('6.5.0_ok.mjs', async () => {
        const filePath = '6.5.0_ok.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toStrictEqual([{
            filePath,
            messages: [],
        }]);
    });

    test('6.5.0_error.mjs', async () => {
        const filePath = '6.5.0_error.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toMatchSnapshot();
    });
});

describe('ESLint 6.6.0', () => {
    test('6.6.0_ok.mjs', async () => {
        const filePath = '6.6.0_ok.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toStrictEqual([{
            filePath,
            messages: [],
        }]);
    });

    test('6.6.0_error.mjs', async () => {
        const filePath = '6.6.0_error.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toMatchSnapshot();
    });
});

describe('ESLint 6.7.0', () => {
    test('6.7.0_ok.mjs', async () => {
        const filePath = '6.7.0_ok.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toStrictEqual([{
            filePath,
            messages: [],
        }]);
    });

    test('6.7.0_error.mjs', async () => {
        const filePath = '6.7.0_error.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toMatchSnapshot();
    });
});
