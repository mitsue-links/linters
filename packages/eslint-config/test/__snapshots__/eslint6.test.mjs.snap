// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ESLint 6.2.0 6.2.0_error.mjs 1`] = `
[
  {
    "filePath": "6.2.0_error.mjs",
    "messages": [
      {
        "line": 2,
        "ruleId": "Unused eslint-disable directive (no problems were reported from 'function-paren-newline').",
        "severity": 1,
      },
      {
        "line": 2,
        "ruleId": "Unused eslint-disable directive (no problems were reported from 'object-curly-spacing').",
        "severity": 1,
      },
      {
        "line": 2,
        "ruleId": "Unused eslint-disable directive (no problems were reported from 'object-property-newline').",
        "severity": 1,
      },
    ],
  },
]
`;

exports[`ESLint 6.4.0 6.4.0_error.mjs 1`] = `
[
  {
    "filePath": "6.4.0_error.mjs",
    "messages": [
      {
        "line": 5,
        "ruleId": "prefer-regex-literals",
        "severity": 2,
      },
      {
        "line": 10,
        "ruleId": "default-param-last",
        "severity": 2,
      },
    ],
  },
]
`;

exports[`ESLint 6.5.0 6.5.0_error.mjs 1`] = `
[
  {
    "filePath": "6.5.0_error.mjs",
    "messages": [
      {
        "line": 5,
        "ruleId": "use-isnan",
        "severity": 2,
      },
      {
        "line": 12,
        "ruleId": "use-isnan",
        "severity": 2,
      },
    ],
  },
]
`;

exports[`ESLint 6.6.0 6.6.0_error.mjs 1`] = `
[
  {
    "filePath": "6.6.0_error.mjs",
    "messages": [
      {
        "line": 4,
        "ruleId": "use-isnan",
        "severity": 2,
      },
      {
        "line": 5,
        "ruleId": "use-isnan",
        "severity": 2,
      },
      {
        "line": 14,
        "ruleId": "no-unsafe-negation",
        "severity": 2,
      },
    ],
  },
]
`;

exports[`ESLint 6.7.0 6.7.0_error.mjs 1`] = `
[
  {
    "filePath": "6.7.0_error.mjs",
    "messages": [
      {
        "line": 13,
        "ruleId": "grouped-accessor-pairs",
        "severity": 2,
      },
      {
        "line": 26,
        "ruleId": "no-constructor-return",
        "severity": 2,
      },
      {
        "line": 42,
        "ruleId": "no-dupe-else-if",
        "severity": 2,
      },
      {
        "line": 59,
        "ruleId": "no-setter-return",
        "severity": 2,
      },
      {
        "line": 68,
        "ruleId": "prefer-exponentiation-operator",
        "severity": 2,
      },
      {
        "line": 76,
        "ruleId": "no-useless-computed-key",
        "severity": 2,
      },
      {
        "line": 78,
        "ruleId": "no-useless-computed-key",
        "severity": 2,
      },
      {
        "line": 80,
        "ruleId": "no-useless-computed-key",
        "severity": 2,
      },
      {
        "line": 82,
        "ruleId": "no-useless-computed-key",
        "severity": 2,
      },
      {
        "line": 84,
        "ruleId": "no-useless-computed-key",
        "severity": 2,
      },
    ],
  },
]
`;
